### PR TITLE
[PyOV] Fix userdata default value

### PR DESCRIPTION
**Pull request by akuporos**
### Details:
 - `userdata` list hasn't been initialized

### Tickets:
 - 92376

### CODE REVIEW

File: src/bindings/python/src/pyopenvino/core/async_infer_queue.cpp [hunk 1]
```diff
@@ -359,7 +359,7 @@ void regclass_AsyncInferQueue(py::module m) {
             return self._user_ids;
         },
         R"(
-        :return: List of all passed userdata. None if the data wasn't passed yet.
+        :return: List of all passed userdata. List[None] if the data wasn't passed yet.
```
Comment (jiwaszki) [inline src/bindings/python/src/pyopenvino/core/async_infer_queue.cpp:362]:
```suggestion
        :return: List of all passed userdata. List is filled with `None` if the data wasn't passed yet.
```
I would suggest something like this, as `List[Any]` and `List[None]` are virtually the same. I think that suggesting specific value (which is in fact there) is better.

File: src/bindings/python/tests/test_runtime/test_infer_request.py [hunk 2]
```diff
@@ -493,6 +493,15 @@ def callback(request, _):
     infer_queue.wait_all()
 
 
+def test_infer_queue_userdata_is_empty(device):
+    core = Core()
+    param = ops.parameter([10])
+    model = Model(ops.relu(param), [param])
+    compiled_model = core.compile_model(model, device)
+    infer_queue = AsyncInferQueue(compiled_model, 1)
+    assert infer_queue.userdata == [None]
```
Comment (jiwaszki) [inline src/bindings/python/tests/test_runtime/test_infer_request.py:502]: Can we test for more values? Just bump up number of requests.

**Review by jiwaszki — COMMENTED** [review]
Overall looks good to me 🥇 Just small nitpicks.

File: src/bindings/python/tests/test_runtime/test_infer_request.py (hunk 2 above)
Comment (akuporos) [reply src/bindings/python/tests/test_runtime/test_infer_request.py:502]: added

File: src/bindings/python/src/pyopenvino/core/async_infer_queue.cpp (hunk 1 above)
Comment (akuporos) [reply src/bindings/python/src/pyopenvino/core/async_infer_queue.cpp]: applied

**Review by jiwaszki — APPROVED** [review]

**Review by p-wysocki — APPROVED** [review]

**Pull request merged**